### ISSUE TITLE
Soundcloud and YouTube

### ISSUE THREAD

**Issue by nibelungen** [volumio/Volumio2#311]
Hey,

at first thanks for the great work so far on volumio2.
Maybe some one can clarify the current state for the integration of soundcloud 'urls' (supported by mpd since 0.17 and got updated to use https since 0.19) and of youtube 'urls' (maybe with youtube-dl).
Is this on the roadmap of volumio2 too?

Best regards,
-andreas

**Comment by apinner** [comment]
Aren't both of those low quality with very compressed audio? I'm not sure that fits with the whole 'audiophile' idea does it?

**Comment by silopolis** [comment]
@AlphaPapa may be (not even sure), but "what can do more, can do less", and
many of us may not want/be able to have several systems depending on media
quality !

**Comment by apinner** [comment]
That's fair enough, just playing devils advocate really.

**Comment by alphapapa** [comment]
@silopolis What's that you say?

**Comment by nibelungen** [comment]
@apinner Where does low and high quality start and end? Currently i stream via bubbleupnp and mycloudplayer soundcloud urls to an volumio rpi, output via audiojack (in my garden) So i really dont care in this case for high (hifi) quality... Just wanna hear music from different sources, mostly soundcloud at the moment.
So for me - and maybe a lot of other volumio users, it would be a nice feature to use the mpd built in option to play soundcloud urls.

**Comment by volumio** [comment]
To me is fine to add soundcloud... What would be the browsing mechanism for it that you envision?

**Comment by apinner** [comment]
Like I said, just playing Devils advocate! I don't mind if either are added..

Soundcloud might be tricky, their API has rate limits on play requests of 15,000 per day per app. Like you say, its hard to imagine a mechanism to browse a site like soundcloud within volumio. Maybe just the ability to search?

**Comment by nibelungen** [comment]
@volumio Great to hear!
Maybe like the xbmc/kodi plug-in and as @apinner mentioned - just a search and list the results.
Or for a first shot - just a input field to place the soundcloud url (as far is i know, mpd just need the url).

**Comment by nibelungen** [comment]
Maybe helpful http://jeroendoggen.github.io/youtube-to-mpd/

**Comment by TechDark** [comment]
maybe the same layout just like the spotify browsing

**Comment by omaralvarez** [comment]
Any news on this feature? I would love to be able to listen to SoundCloud liked tracks or playlists.

**Comment by Zelaf** [comment]
If you need to log in to soundcloud to be able to browse it you can display artists you follow in a page, then when you enter artists that you follow you can click on songs etc.

Also, would love to see soundcloud.

**Comment by sla89** [comment]
FYI: we are currently implementing a fully integrated youtube plugin for volumio.

**Comment by sepastian** [comment]
@sla89, what's the status on this? Can it be tested already?

**Comment by sla89** [comment]
@sepastian no, not yet. Do you want to test it when we have a stable development version ready like an alpha or beta?

**Comment by sepastian** [comment]
Sure, let me know when I can test.

**Comment by Zelaf** [comment]
If it's not an issue I could also test it. It would get me a chance to work on my bug reporting skills and I would also get more chances to using my Volumio setup :D

**Comment by TechDark** [comment]
So there is a soundcloud plugin?

**Comment by Zelaf** [comment]
@TechDark I dunno, is there?

**Comment by sla89** [comment]
@sepastian and @Zelaf I will come back to you soon 👍 Thanks for your help.
@TechDark as far as I know there is no plugin for soundcloud

**Comment by crisp00** [comment]
Hi everyone! 

The YouTube plugin that _sla89_ and _I_ have been working on has finally been pushed to beta!!! You can find it in your __Volumio UI__ ,  _Plugins_ ->_Search Plugins_ ->_Music Services_ ! 

For bug reports and requests the best place is the volumio forums, where we have an official thread. You can also post an issue on my repo  __PiYoutube__  if you prefer.

A SoundCloud plugin may be coming next! Good things happen to those who wait!

Best, Cris Pintea.

**Comment by omaralvarez** [comment]
I'm eagerly awaiting for the Soundcloud plugin. Godspeed and thanks my friend!

**Comment by volderette** [comment]
Soundcloud would be my most awaited feature as well.

**Comment by BenjaminKauer** [comment]
Soundcloud support shouldn't be that much of a problem, if we make the API key configurable by the user. It'd be the users job to receive a Soundcloud API key (which is currently not possible, btw).

See also: [forum post "SOUNDCLOUD SUPPORT"](https://volumio.org/forum/soundcloud-support-t3059.html)

**Comment by frederiktoft** [comment]
Just want to put out a request for Soundcloud support as well. Mopidy supports it with personal api key, so it should be feasible to do a Volumio SoundCloud plug-in using the same method.

**Comment by yakir12** [comment]
Me too!

**Comment by dmnBrest** [comment]
+1 for Soundcloud

**Comment by gruentee** [comment]
Yes, SoundCloud support would be really great.

**Comment by w97** [comment]
please make this work, thank you! :)

**Comment by Pejosonic** [comment]
I would be great to have a Soundcloud integration!!

**Comment by mujahed1987** [comment]
Soundclound integration will be great addition to Volumio

**Comment by hussong** [comment]
SoundCloud audio quality is pretty good with a SoundCloud Go+ account (10 €/mo), then you can stream most tracks in 256k AAC (some are still only 128k MP3, I guess this is because the original upload wasn't higher quality). It is also possible to download these streams with a patched version of youtube-dl. 

I would really enjoy a SoundCloud integration as well!

**Comment by cmosek** [comment]
+1 for Soundcloud integration!

**Comment by Pejosonic** [comment]
Hi, i really would love to find the way to listen Souncloud tracks on my Volumio player.
Is there some workaround to do this in the current Volumio version?

Is there a particullar reason why the team it's not taking place for this feature request?
Maybe we could take some inspiration for the browsing from the Soundcloud Kodi add-on?

Thank you!
Regards

**Comment by Titaantje** [comment]
Soundcloud integration, is it there yet?? 
Even the most basic integration like an url plugion would be BOMB :)
